test/apps/iodemo/run_io_demo.sh

```diff
@@ -0,0 +1,456 @@
+			# For each node, add both servers and clients, the last node may
+			# have less than others
+			node_num_servers=$(min ${max_servers_per_node} $((num_servers - added_servers)))
+			node_num_clients=$(min ${max_clients_per_node} $((num_clients - added_clients)))
+			;;
+		slot)
+			# Servers take slots first, and cleints take what is left
+			node_num_servers=$(min ${tasks_per_node} $((num_servers - added_servers)))
+			node_num_clients=$(min $((tasks_per_node - node_num_servers)) \
+			                       $((num_clients - added_clients)))
+			;;
+		esac
+
+		num_servers_per_host[${host}]=${node_num_servers}
+		num_clients_per_host[${host}]=${node_num_clients}
+
+		for ((i=0;i<node_num_servers;++i))
+		do
+			port_num=$((base_port_num + i))
+			client_connect_list+=" ${ip_address_per_host[${host}]}:${port_num}"
+		done
+
+		added_servers=$((added_servers + node_num_servers))
+		added_clients=$((added_clients + node_num_clients))
+	done
+
+	#
+	# In verbose mode, make following changes:
+	#   1. show mixed output in stdout (in addition to log files)
+	#   2. run the spawned scripts with '-x'
+	#   3. show output from kill commands
+	#
+	if is_verbose
+	then
+		show_var client_connect_list
+		sh_args="-x"
+		wait_redirect=""
+		log_redirect="|& tee "
+	else
+		sh_args=""
+		wait_redirect=">& /dev/null"
+		log_redirect=">& "
+	fi
+
+	# create run scripts
+	for host in $(split_list ${host_list})
+	do
+		command_file="iodemo_commands_${host}.sh"
+
+		# Add file header and startup
+		cat >${command_file} <<-EOF
+			#!/bin/sh ${sh_args}
+			#
+			# Launch script for io_demo on ${host} with ${num_servers_per_host[${host}]} servers and ${num_clients_per_host[${host}]} clients
+			#
+
+			kill_iodemo() {
+			    # kill process by exact name to avoid killing launch scirpts
+			    # lt- prefix is in case iodemo is wrapped by libtool
+			    pkill --exact -u ${USER} -9 $(basename ${iodemo_exe})
+			    pkill --exact -u ${USER} -9 lt-$(basename ${iodemo_exe})
+			}
+
+			signal_handler() {
+			    echo "Got signal, killing all iodemo processes"
+			    kill_iodemo
+			    exit
+			}
+
+			# kill existing processes and trap signals
+			trap kill_iodemo INT TERM
+			kill_iodemo
+
+			# Use same work dir as launch script
+			cd $PWD
+
+			# UCX inherited environment variables
+			EOF
+
+		# Add all env vars which start with UCX_ to the command file
+		env | grep '^UCX_' | xargs -L 1 echo export >>${command_file}
+
+		# Add servers commands
+		cat >>${command_file} <<-EOF
+
+			# Server commands
+			EOF
+
+		for ((i=0;i<${num_servers_per_host[${host}]};++i))
+		do
+			port_num=$((base_port_num + i))
+			log_file=$(printf "iodemo_%s_server_%02d.log" ${host} $i)
+			echo ${log_file}
+			cat >>${command_file} <<-EOF
+				stdbuf -e0 -o0 ${iodemo_exe} ${iodemo_server_args} -p ${port_num} ${log_redirect} ${log_file} &
+				EOF
+		done
+
+		# Add short sleep to let servers start running
+		cat >>${command_file} <<-EOF
+
+			# Wait for servers to start
+			sleep 2
```

Comment:
maybe s/2/$wait_time_sec ? and define it at the script top in case 2 will be not enough?